AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -18,76 +18,129 @@ extension Model {
-            if case .collection = field.type {
+            if case .collection = modelField.type {
                 return
             }
 
-            let name = field.graphQLName
-            let fieldOptionalValue: Any??
-
-            if let jsonModel = self as? JSONValueHolder {
-                fieldOptionalValue = jsonModel.jsonValue(for: field.name, modelSchema: modelSchema) ?? nil
-            } else {
-                fieldOptionalValue = self[field.name] ?? nil
-            }
+            let fieldValueOptional = getFieldValue(for: modelField.name, modelSchema: modelSchema)
 
             // Since the returned value is Any?? we need to do the following:
             // - `guard` to make sure the field name exists on the model
             // - `guard` to ensure the returned value isn't nil
-            guard let fieldValue = fieldOptionalValue else {
-                input.updateValue(nil, forKey: name)
+            guard let fieldValue = fieldValueOptional else {
+                input.updateValue(nil, forKey: modelField.name)
                 return
             }
 
             // swiftlint:disable:next syntactic_sugar
             guard case .some(Optional<Any>.some(let value)) = fieldValue else {
-                input.updateValue(nil, forKey: name)
+                input.updateValue(nil, forKey: modelField.name)
                 return
             }
 
-            switch field.type {
+            switch modelField.type {
             case .date, .dateTime, .time:
                 if let date = value as? TemporalSpec {
-                    input[name] = date.iso8601String
+                    input[modelField.name] = date.iso8601String
                 } else {
-                    input[name] = value
+                    input[modelField.name] = value
                 }
             case .enum:
-                input[name] = (value as? EnumPersistable)?.rawValue
+                input[modelField.name] = (value as? EnumPersistable)?.rawValue
             case .model:
-                // For Models, append the model name in front in case a targetName is not provided
-                // e.g. "comment" + "PostId"
-                var fieldName = modelName.camelCased() + name
-                if case let .belongsTo(_, targetName) = field.association {
-                    fieldName = targetName ?? fieldName
-                }
-                if let modelValue = value as? Model {
-                    input[fieldName] = modelValue.id
-
-                } else if let value = value as? [String: JSONValue],
-                          case .string(let primaryKeyValue) = value[modelSchema.primaryKey.name] {
-                    input[fieldName] = primaryKeyValue
-                } else {
-                    input[fieldName] = nil
-                }
+                let fieldName = getFieldNameForAssociatedModels(modelField: modelField) ?? modelField.name
```

Comment:
tiny nit: If the correct behavior is always to use the `modelField.name`, would it make sense to make that the return value from `getFieldNameForAssociatedModels`? Or is `nil` a meaningful value in that case?

Reply:
actually the previous code is `var fieldName = modelName.camelCased() + name` (name is resolved earlier as `name + "Id"`) which means it will always resolve to the default "postCommentId" for the associated models. I don't believe there's a case for `getFieldNameForAssociatedModels` to return nil for it to take `modelField.name`, i will update this to remove `modelField.name`